numpyro/svi.py

```diff
@@ -151,7 +155,15 @@ def elbo(param_map, model, guide, model_args, guide_args, kwargs, constrain_fn):
     """
     param_map = constrain_fn(param_map)
     guide_log_density, guide_trace = log_density(guide, guide_args, kwargs, param_map)
-    model_log_density, _ = log_density(replay(model, guide_trace), model_args, kwargs, param_map)
+    # NB: we only want to substitute params not available in guide_trace
+    param_map = {k: v for k, v in param_map if k not in guide_trace}
+    # NB: only skip transforms for AutoContinuous guide
+    skip_dist_transforms = False
+    guide_fn = guide.fn if isinstance(guide, Messenger) else guide
+    if isinstance(guide_fn, AutoContinuous):
+        skip_dist_transforms = True
+    model_log_density, _ = log_density(replay(model, guide_trace), model_args, kwargs, param_map,
+                                       skip_dist_transforms=skip_dist_transforms)
```

Comment:
This is where I found the trickiest. I might be making a mistake here. In autocontinuous, we only adjust jacobian from unconstrained_value -> base_value, hence I think that for log_density, we only need to compute log_prob of base distributions.

Reply:
This seems right to me reg. the jacobian adjustment term. The other issue that we discussed reg. having `param` statements with dynamic parameters should just be handled by your changes to `svi.init_fn` and the `substitute` handler. We can probably just test this with SVI on `test_improper_prior` by moving the param statements to a guide and sampling from the Normal distribution and changing the param statements in the model to sample statements with suitable priors.

Reply:
Thanks @neerajprad ! I just add a test for SVI, it works well. Currently, the test for autoguide fails so I'll keep debugging for it.